pkg/plugin/response.go

```diff
@@ -277,3 +298,132 @@ func labelsToJSON(labels data.Labels) (json.RawMessage, error) {
 
 	return b, nil
 }
+
+const (
+	vector, matrix, scalar = "vector", "matrix", "scalar"
+)
+
+// Result represents timeseries from query
+type Result struct {
+	Labels Labels  `json:"metric"`
+	Values []Value `json:"values"`
+	Value  Value   `json:"value"`
+}
+
+// Value represents timestamp and value of the timeseries
+type Value [2]interface{}
+
+// Labels represents timeseries labels
+type Labels map[string]string
+
+// Data contains fields
+// ResultType which defines type of the query response
+// Result resents json of the query response
+type Data struct {
+	ResultType string          `json:"resultType"`
+	Result     json.RawMessage `json:"result"`
+}
+
+// Response contains fields from query response
+type Response struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+
+type promInstant struct {
+	Result []Result `json:"result"`
+}
+
+func (pi promInstant) dataframes() (data.Frames, error) {
+	frames := make(data.Frames, len(pi.Result))
+	for i, res := range pi.Result {
+		f, err := strconv.ParseFloat(res.Value[1].(string), 64)
+		if err != nil {
+			return nil, fmt.Errorf("metric %v, unable to parse float64 from %s: %w", res, res.Value[1], err)
+		}
+
+		ts := time.Unix(int64(res.Value[0].(float64)), 0)
+		frames[i] = data.NewFrame("",
+			data.NewField(data.TimeSeriesTimeFieldName, nil, []time.Time{ts}),
+			data.NewField(data.TimeSeriesValueFieldName, data.Labels(res.Labels), []float64{f}))
+	}
+
+	return frames, nil
+}
+
+type promRange struct {
+	Result []Result `json:"result"`
+}
+
+func (pr promRange) dataframes() (data.Frames, error) {
+	frames := make(data.Frames, len(pr.Result))
+	for i, res := range pr.Result {
+		timestamps := make([]time.Time, len(res.Values))
+		values := make([]float64, len(res.Values))
+		for j, value := range res.Values {
+			v, ok := value[0].(float64)
+			if !ok {
+				return nil, fmt.Errorf("error get time from dataframes")
+			}
+			timestamps[j] = time.Unix(int64(v), 0)
+
+			f, err := strconv.ParseFloat(value[1].(string), 64)
+			if err != nil {
+				return nil, fmt.Errorf("erro get value from dataframes: %s", err)
+			}
+			values[j] = f
+		}
+
+		if len(values) < 1 || len(timestamps) < 1 {
+			return nil, fmt.Errorf("metric %v contains no values", res)
+		}
+
+		frames[i] = data.NewFrame("",
+			data.NewField(data.TimeSeriesTimeFieldName, nil, timestamps),
+			data.NewField(data.TimeSeriesValueFieldName, data.Labels(res.Labels), values))
+	}
+
+	return frames, nil
+}
+
+type promScalar Value
```

Comment:
vlogs doesn't return scalar types in stats response - see here https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app/vlselect/logsql

Reply:
removed it.